extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CacheManagerInitializer.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.cache.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import io.quarkus.cache.CacheManager;
+
+/**
+ * This class is used to eagerly create the {@link CacheManager} bean instance at STATIC_INIT execution time.
+ */
+@ApplicationScoped
```

Comment:
Hm, since the only purpose of this bean is to initialize the cache manager I'd recommend to use the `@Dependent` scope instead (in fact you can skip the scope annotation because `@Dependent` is added automatically if a class declared an observer method but no bean defining annotation).  `@ApplicationScoped` bean instance stays in the context for the lifetime of the application (unless explicitly removed).

Furthermore, you can inject the `CacheManager` as the `onStaticInit()` observer method parameterer and then you can mark the observer method `static` because no instance members are needed.

Reply:
Thanks for the suggestions, it looks much better now :)

Reply:
You're welcome! :-)